server/api/rounds.py

```diff
@@ -724,10 +724,36 @@ def validate_round(round: dict, election: Election):
         if "sampleSizes" not in round:
             raise BadRequest("Sample sizes are required for round 1")
 
-        targeted_contest_ids = {c.id for c in election.contests if c.is_targeted}
-        if set(round["sampleSizes"].keys()) != targeted_contest_ids:
+        targeted_contests = [
+            contest for contest in election.contests if contest.is_targeted
+        ]
+        if set(round["sampleSizes"].keys()) != {c.id for c in targeted_contests}:
             raise BadRequest("Sample sizes provided do not match targeted contest ids")
 
+        if election.audit_type == AuditType.BATCH_COMPARISON:
+            for single_contest in targeted_contests:
+                total_batches = sum(
+                    jurisdiction.manifest_num_batches or 0
+                    for jurisdiction in single_contest.jurisdictions
+                )
+                if round["sampleSizes"][single_contest.id] > total_batches:
+                    raise Conflict(
+                        f"Sample size must be less than or equal to: {total_batches} (the total number of batches in the targeted contest)"
+                    )
+
+        if (
+            election.audit_type == AuditType.BALLOT_POLLING
+            or election.audit_type == AuditType.BALLOT_COMPARISON
+        ):
+            for single_contest in targeted_contests:
+                if (
+                    round["sampleSizes"][single_contest.id]
+                    > single_contest.total_ballots_cast
+                ):
+                    raise Conflict(
+                        f"Sample size must be less than or equal to: {single_contest.total_ballots_cast} (the total number of ballots in the targeted contest)"
+                    )
```

Comment:
might be helpful to include the contest name in the error message so the user knows which contest they need to correct the sample size for

Reply:
the error message for batch comparison also needs this change